packages/mermaid/src/rendering-util/getTextWidth.js

```diff
@@ -0,0 +1,7 @@
+// https://stackoverflow.com/a/35373030/3469145
+const canvas = document.createElement('canvas');
+const context = canvas.getContext('2d');
+
+const getTextWidth = (text) => context.measureText(text).width * window.devicePixelRatio;
+
+export { getTextWidth };
```

Comment:
https://github.com/mermaid-js/mermaid/blob/28406fc9c4ff0a8426d4f08d0395630b39955aee/packages/mermaid/src/rendering-util/createText.ts#L70-L77

There is another width calculation that we use, which uses SVG instead of canvas.
As everything else in mermaid is SVG, I think we should use SVG based width calculation, so the correct styles (font, font size, etc) is applied to the calculation.

Reply:
Hi, I just pushed related changes, I'm not familiar with d3, hope I got this right...

Reply:
The changes look good, does it work for you locally?
Can you add an integration test to ensure that the correct attribute was added to the final SVG?